surefire-its/src/test/java/org/apache/maven/surefire/its/RunOrderIT.java

```diff
@@ -97,7 +97,17 @@ public void testRandomJUnit4SameSeed()
             }
         }
     }
-    
+
+    @Test
+    public void testRandomJUnit4PrintSeed()
+    {
+        long seed = 0L;
+        OutputValidator validator = executeWithRandomOrder( "junit4", seed );
+        validator.verifyTextInLog( "To reproduce ordering use flag" );
+        validator = executeWithRandomOrder( "junit4" );
+        validator.verifyTextInLog( "To reproduce ordering use flag" );
```

Comment:
please consider to split into two test, eg:
- testRandomJUnit4PrintSeedWithGivenSeed
- testRandomJUnit4PrintSeedWithNoGivenSeed

Reply:
Hi @delanym . Nice to meet you after the release Vote!
I know that it is a matter of taste but these names of classes and methods determine the length of folders on the file system in our CI. Using `WithSeed` and `WithoutSeed` is shorter.
Now, I have realized that @delanym is using two executions which might be utilized in some way.
@delanym what wat your point? You wanted to compare the XML reports and the order of tests? Something else? If you want to make sure that the seed is set, such situation can be verified with the old IT where a debug message can be printed in the logs and the IT can check that the message was printed with seed != NULL.

Reply:
@delanym 
We are doing these things twice. The seed is set to `System.nanoTime()` if it was NULL, see `DefaultRunOrderCalculator`.
This means that the only work here is to change the condition in the IF statement but nothing more!
```
        Long runOrderRandomSeed = runOrderParameters.getRunOrderRandomSeed();
        if ( runOrderRandomSeed == null )
        {
            runOrderRandomSeed = System.nanoTime();
            runOrderParameters.setRunOrderRandomSeed( runOrderRandomSeed );
        }
        this.random = new Random( runOrderRandomSeed );
```

Reply:
@slawekjaranowski Do you remember all the years we solved together with Robert and Infra team because the Windows paths were limited on the file system? They are!

Reply:
@delanym I made `RunOrderParameters` immutable because the contributor did not implement it right and the setter was not necessary. You should exclude the fallback value `System.nanoTime()` in the `DefaultRunOrderCalculator` constructor and use it only in MOJO. It does not make sense to do one thing twice with the seed.

Reply:
Have I done that right?